### PR TITLE
Add altinn validation before a salary can be locked

### LINKED ISSUE #176: Validate salary before lock
2016-06-01_11-53-39_validation_salary_on_lock

### PR DESCRIPTION
**Pull request by mladjo2505**
Closes #176

### CODE REVIEW

**Comment by sls-lnluksa** [comment]
First round review: Nikola Lukovic (sls-lnluksa)
Final review with: @stoivo 

Printing errors is not on Lock, but whenever we come to page. if we have some errors Lock button is disabled.
Same way of printing error is on invoiceout.edit. 

We think this is ok way.

**Comment by stoivo** [comment]
Altinndato can be in a different period than the salary period.

if altinn dato is not set only warn about that. Don't check anything else.

File: modules/salary/view/edit.php [hunk 1]
```diff
@@ -562,7 +563,11 @@
...
             $altinn_arguments = "'$head->ShiftType', '$head->WorkTimeScheme', '$head->TypeOfEmployment', $head->OccupationID, $head->SubcompanyID, $is_altinn_date_set";
-            if(!$head->LockedBy) echo '<input type="submit" name="action_salary_lock" value="L&aring;s (L)" accesskey="L" onclick="return checkIfAltinnFieldsSetAndConfirm(\'Er du siker?\', '.$altinn_arguments.');" />';
+            $report_for_salary = new altinn_report($head->Period, array($head->SalaryID), array($head->WorkRelationID), false);
+            $report_for_salary->populateReportArray();
+            $no_altinn_validation_errors = empty($report_for_salary->errors);
+            if (!$no_altinn_validation_errors) echo "<div class='warning'>" . implode($report_for_salary->errors, "<br/>") . "</div>";
+            if(!$head->LockedBy) echo '<input type="submit" name="action_salary_lock" value="L&aring;s (L)" accesskey="L" onclick="return checkIfAltinnFieldsSetAndConfirm(\'Er du siker?\', '.$altinn_arguments.');" ' . (($no_altinn_validation_errors) ? '' : 'disabled') . ' />';
```
Comment (stoivo) [inline modules/salary/view/edit.php]: Add colspan.

Comment (stoivo) [inline modules/salary/view/edit.php]: Add colspan to td.

**Comment by stoivo** [comment]
We need to be able to lock salary before we have altinn date.

**Comment by mladjo2505** [comment]
Not good if we have more salaries in the period that the selected salary is in.
If Altinn date is not set warn about that also.
Only if there are no errors to send to altinn enable lock even if there is no altinn date set.

**Comment by sls-lnluksa** [comment]
First round review: Nikola Lukovic (sls-lnluksa)
Final review with: @stoivo

File: modules/salary/view/edit.php [hunk 2]
```diff
@@ -558,11 +559,18 @@
...
+          <td colspan='14'>
           <?php
             $is_altinn_date_set = (is_null($head->ActualPayDate) || $head->ActualPayDate == '' || $head->ActualPayDate == '0000-00-00') ? 'false' : 'true';
             $altinn_arguments = "'$head->ShiftType', '$head->WorkTimeScheme', '$head->TypeOfEmployment', $head->OccupationID, $head->SubcompanyID, $is_altinn_date_set";
-            if(!$head->LockedBy) echo '<input type="submit" name="action_salary_lock" value="L&aring;s (L)" accesskey="L" onclick="return checkIfAltinnFieldsSetAndConfirm(\'Er du siker?\', '.$altinn_arguments.');" />';
+            $report_for_salary = new altinn_report($head->Period, array($head->SalaryID), array($head->WorkRelationID), false, true);
+            $report_for_salary->populateReportArray();
+            $no_altinn_validation_errors = empty($report_for_salary->errors);
+            $altinn_validations_error_array = (($no_altinn_validation_errors) ? array() : $report_for_salary->errors);
+            $altinndate_missing_error_array = ($head->ActualPayDate == '0000-00-00' || empty($head->ActualPayDate)) ? array("Altinndato ikke satt") : array();
```
Comment (mladjo2505) [inline modules/salary/view/edit.php]: `Altinndato er ikke satt`

File: modules/salary/view/edit.php [hunk 3]
```diff
@@ -558,11 +559,18 @@
...
+            $errors = array_merge($altinndate_missing_error_array, $altinn_validations_error_array);
+            if ($errors) echo "<div class='warning'>" . implode($errors, "<br/>") . "</div>";
+            if(!$head->LockedBy) echo '<input type="submit" name="action_salary_lock" value="L&aring;s (L)" accesskey="L" onclick="return checkIfAltinnFieldsSetAndConfirm(\'Er du siker?\', '.$altinn_arguments.');" ' . (($no_altinn_validation_errors) ? '' : 'disabled') . ' />';
```
Comment (mladjo2505) [inline modules/salary/view/edit.php]: `Er du sikker?`

**Pull request merged**